server/app/repository/VersionRepository.java

```diff
@@ -581,12 +606,17 @@ private static ImmutableMap<Long, String> getQuestionIdToNameMap(Version version
    */
   private static ImmutableSet<String> getProgramQuestionNames(
       ProgramDefinition program, ImmutableMap<Long, String> questionIdToNameLookup) {
+    return getQuestionsInProgram(program).stream()
+        .filter(questionIdToNameLookup::containsKey)
+        .map(questionIdToNameLookup::get)
+        .collect(ImmutableSet.toImmutableSet());
+  }
+
+  private static ImmutableList<Long> getQuestionsInProgram(ProgramDefinition program) {
```

Comment:
**suggestion (blocking):** implement this on `ProgramDefinition` instead

Reply:
Updated.